test/reporters/helpers.js

```diff
@@ -0,0 +1,119 @@
+'use strict';
+
+/*
+  This function prevents the constant use of creating a runnerEvent.
+  runStr is the argument that solely defines the runnerEvent.
+  ifStr1 is one possible reporter argument, as is ifStr2, and ifStr3
+  arg1 and arg2 are the possible variables that need to be put into the scope of this function for the tests to run properly.
+*/
+function runnerEvent (runStr, ifStr1, ifStr2, ifStr3, arg1, arg2) {
+  var test = null;
+  if (runStr === 'start' || runStr === 'pending' || runStr === 'end') {
+    return function (event, callback) {
+      if (event === ifStr1) {
+        callback();
+      }
+    };
+  } else if (runStr === 'pending test' || runStr === 'pass' || runStr === 'fail' || runStr === 'end' || runStr === 'suite' || runStr === 'suite end' || runStr === 'test end') {
```

Comment:
It's hard to read and makes a verbose diff when we add a new case.
What do you think about format it like:

```
} else if (
  runStr === "pending test" ||
  runStr === "pass" ||
  runStr === "fail" ||
  runStr === "end" ||
  runStr === "suite" ||
  runStr === "suite end" ||
  runStr === "test end"
) {
```